### PR TITLE
Lit: Add typings for @storybook/lit

### PR DESCRIPTION
**Pull request by yannbf**
Issue: Could be part of #14930 or I can create a new one

## What I did
1 - Add typings for `Story` and `Meta`
2 - Stop exposing internal types that are used for ClientApi

## How to test

- Is this testable with Jest or Chromatic screenshots? no
- Does this need a new example in the kitchen sink apps? yes, it's updated!
- Does this need an update to the documentation? no

If your answer is yes to any of these, please make sure to include it in your PR.

## Current limitation

It would be amazing if we could just automatically extract the properties of a given component, like we do with Angular components. I'm unsure whether this is already possible but I've [filed an issue](https://github.com/lit/lit/issues/1902) and hopefully we can get some perspective on this!

### CODE REVIEW

**Comment by urish** [comment]
Not sure if this is the right place to ask, but is @storybook/lit useful yet? 

I'm looking to upgrade [wokwi-element](https://github.com/wokwi/wokwi-elements) from lit-element to lit, and right now storybook support is the only blocker (wokwi/wokwi-elements#67).

File: examples/lit-kitchen-sink/src/stories/addons/controls/addon-controls.stories.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { html } from 'lit';
+import { Story, Meta } from '@storybook/lit';
 import '../../../components/SbButton';
+import { MetaProperty } from 'typescript';
```
Comment (shilman) [inline examples/lit-kitchen-sink/src/stories/addons/controls/addon-controls.stories.ts:4]:
```suggestion
```

File: examples/lit-kitchen-sink/src/stories/addons/controls/addon-controls.stories.ts [hunk 2]
```diff
@@ -12,9 +14,9 @@ export default {
       },
     },
   },
-};
+} as Meta;
 
-const Template = ({ primary, backgroundColor, size, label, sbButtonClickHandler }: any) =>
+const Template: Story = ({ primary, backgroundColor, size, label, sbButtonClickHandler }) =>
```
Comment (shilman) [inline examples/lit-kitchen-sink/src/stories/addons/controls/addon-controls.stories.ts:19]: `Story<???>`?

**Review by shilman — APPROVED** [review]
LGTM tho the types could be stronger, e.g. `Meta<X>`/`Story<X>`

Plus one typo (editor auto-import)?

**Comment by yannbf** [comment]
> LGTM tho the types could be stronger, e.g. `Meta<X>`/`Story<X>`
> 
> Plus one typo (editor auto-import)?

The auto import was totally a mistake hehe but it's fixed. `About Meta<X>`, given that I couldn't find a way to extract properties from `LitElement` type, Args for lit will be more manual until I get some feedback in the issue I mentioned in the description of this PR.. in that case, in the kitchen sink examples we can create and export the types to use like `Story<Props>`. 

@gaetanmaisse what would be the preferred way? Creating a type together with the component and using in the stories.. or creating the type directly in the story file? I assume people don't do this often because the module augmentation (HTMLElementTagNameMap) in the component file takes care of types when writing the web component, right? People don't really need to specify the props anywhere.. but in Storybook they would, unless we could extract the types out of the component class like it's done in Angular.

**Comment by shilman** [comment]
@yannbf OK merging this now and let's figure out the stricter typing in a follow-up!

**Pull request merged**